### PR TITLE
Invariants take2 cleanup

### PR DESCRIPTION
**Pull request by grandizzy**

### CODE REVIEW

File: tests/INVARIANTS.md [hunk 1]
```diff
@@ -34,7 +34,9 @@
 - **B2**: bucket LPs accumulator (`Bucket.lps`) = 0 if no deposit / collateral in bucket  
 - **B3**: if no collateral or deposit in bucket then the bucket exchange rate is `1e27`  
 - **B4**: bankrupt bucket LPs accumulator = 0; lender LPs for deposits before bankruptcy time = 0  
-- **B5**: when adding quote tokens: lender deposit time (`Lender.depositTime`) = timestamp of block when deposit happened (`block.timestamp`)  
+- **B5**: when adding / moving quote tokens or adding collateral : lender deposit time (`Lender.depositTime`) = timestamp of block when deposit happened (`block.timestamp`)  
+- **B6**: when receiving transferred LPs : receiver deposit time (`Lender.depositTime`) = max of sender and receiver deposit time
```
Comment (prat-gpt) [inline tests/INVARIANTS.md:38]: these invariants are checked in current `invariant_Bucket_deposit_time_B5`, we can rename this method to `invariant_Bucket_deposit_time_B5_B6_B7`

File: tests/forge/ERC20Pool/invariants/base/BaseHandler.sol [hunk 2]
```diff
@@ -114,13 +108,10 @@ abstract contract BaseHandler is Test {
      * @dev Resets all local states before each action.
      */
     modifier resetAllPreviousLocalState() {
```
Comment (prat-gpt) [inline tests/forge/ERC20Pool/invariants/base/BaseHandler.sol]: can be renamed to `updateLocalStateAndPoolInterest`

File: tests/forge/ERC20Pool/invariants/base/BaseHandler.sol [hunk 3]
```diff
@@ -224,36 +216,21 @@ abstract contract BaseHandler is Test {
...
-        // reset the reserves before each action 
-        previousReserves = 0;
-        currentReserves  = 0;
-    }
-
-    /**
-     * @dev Precalculate exchange rate before an action.
-     */
-    function _updatePreviousExchangeRate() internal {
+    function _recordReservesAndExchangeRate() internal {
```
Comment (prat-gpt) [inline tests/forge/ERC20Pool/invariants/base/BaseHandler.sol]: can be renamed to `_recordAndResetReservesAndExchangeRate`

File: tests/INVARIANTS.md (hunk 1 above)
Comment (grandizzy) [reply tests/INVARIANTS.md:38]: updated with https://github.com/ajna-finance/contracts/pull/682/commits/55d320d54c0b246be9219e2bccbfdefc80f98058

File: tests/forge/ERC20Pool/invariants/base/BaseHandler.sol (hunk 2 above)
Comment (grandizzy) [reply tests/forge/ERC20Pool/invariants/base/BaseHandler.sol]: renamed with https://github.com/ajna-finance/contracts/pull/682/commits/55d320d54c0b246be9219e2bccbfdefc80f98058

File: tests/forge/ERC20Pool/invariants/base/BaseHandler.sol (hunk 3 above)
Comment (grandizzy) [reply tests/forge/ERC20Pool/invariants/base/BaseHandler.sol]: changed with https://github.com/ajna-finance/contracts/pull/682/commits/55d320d54c0b246be9219e2bccbfdefc80f98058 though used `_resetAndRecordReservesAndExchangeRate` naming

**Review by prat-gpt — APPROVED** [review]
LGTM! thanks for making the code look far more cleaner than before!

**Pull request merged**